memcache/pool.go

```diff
@@ -20,6 +20,9 @@ type PoolConfig struct {
 	MaxConns        int           // maximum number of connections to keep open per server (includes active and idle connections)
 	MaxIdleLifetime time.Duration // maximum amount of time an idle connection is kept open
 	IdleClosePeriod time.Duration // how often to close (reap) idle connections
+
+	OnAcquire func(string, error, time.Duration) // when we acquire a connection slot
+	OnCreate  func(string, error, time.Duration) // when we create a new connection
```

Comment:
don't love these on the config, I'd rather see it as functions on `Pool`

Reply:
Doesn't really work because there's multiple Pools (one for each node in the cluster). We create a `ClusterPool` to abstract this.

Reply:
🤔 great point!